### ISSUE TITLE
Background (model) migration job

### ISSUE THREAD

**Issue by khenderick** [openvstorage/framework#1118]
At this moment, the upgrade supports immediate migration that takes place during the upgrade. If it would fail, so would the upgrade.

However, there might be stuff that might be possible after an upgrade that is not blocking if it fails, but adds value to the system. A few examples:
* When a certain setting is migrated from volumedriver config file to cluster registry
* When an additional field is added to the model containing some extra information (e.g. disk serial number)
* When additional values are available (e.g. more albadisk aliasses to be loaded from the asdmanager)

Due to the nature of the system, such items might take some time to fully complete; for example, to migrate a volumedriver configuration file, we might have to wait until all volumedriver nodes are upgraded. Or we depend on albanodes to be online.

We need some kind of mechanism that can do these async upgrades in the background:
* Some model version number _per object_ so we can enhance objects as data is available (but don't process them twice)
* Some knowledge about prerequisites when the migration can happen.

Basically, we need in total 3 migration flows:
1. Immediate code migration (before any services are started - config file migration, cleaning old code, ...)
2. Immadiate model migration (after starting model services - make sure the model fields are migrated where required)
3. Async migrations (when everything is running, might take a while)

**Comment by wimpers** [comment]
We should also put the services in some kind of maintenance in case the autoupdate doesn't work .
